bin/nim-gdb

```diff
@@ -5,17 +5,25 @@ set -e
 
 which nim > /dev/null || (echo "nim not in PATH"; exit 1)
 which gdb > /dev/null || (echo "gdb not in PATH"; exit 1)
+which readlink > /dev/null || which greadlink > /dev/null || (echo "readlink not in PATH. Please install coreutils from brew if on Mac."; exit 1)
+
+nreadlink () {
+    (which greadlink > /dev/null && greadlink "$@") || (which readlink > /dev/null && readlink "$@") || echo "Readlink could not be found"
+}
 
 # Find out where the pretty printer Python module is
-NIM_SYSROOT=$(dirname $(dirname $(readlink -e $(which nim))))
+NIM_SYSROOT=$(dirname $(dirname $(nreadlink -e $(which nim))))
 GDB_PYTHON_MODULE_PATH="$NIM_SYSROOT/tools/nim-gdb.py"
 
 # Run GDB with the additional arguments that load the pretty printers
 # Set the environment variable `NIM_GDB` to overwrite the call to a
 # different/specific command (defaults to `gdb`).
 NIM_GDB="${NIM_GDB:-gdb}"
+
+# This is needed for some reason. I can't get -eval-command to work ever
+echo "source $GDB_PYTHON_MODULE_PATH" > gdbcommands.txt
```

Comment:
can you elaborate on this? Why is this needed? What version of gdb are you using (``gdb --version``)? Is it related to an outdated version of ``gdb``?

Reply:
It could have been. I believe I was using 8.3 at the time, but I could be wrong. Just tried with gdb 8.3.1 and it works without this hack.

Reply:
Hmm, it looks like my issue was with VSCode. Using your original code makes VSCode error out with:

```
Loading Nim Runtime support.
('set Nim pretty printers for ', '/Users/joey/Projects/nim-nanovg/build/demosdl2/demo_sdl2_debug.dSYM/Contents/Resources/DWARF/demo_sdl2_debug')
undefinedBFD: /usr/lib/dyld(i386:x86-64): unknown load command 0x34
BFD: /usr/lib/dyld(i386:x86-64): unknown load command 0x34
BFD: /usr/lib/dyld(i386:x86-64): unknown load command 0x34
No source file named /Users/joey/Projects/nim-nanovg/examples/demo_sdl2_gpu.nim.
Running executable
[New Thread 0xf03 of process 2992]
```

I'm not sure what to do with that :/

Reply:
Hmm, it looks like it works if I rename `nim-gdb` to `nim-gdb.bash`. Would that be acceptable to do, and then have a symbolic link from `nim-gdb` to `nim-gdb.bash`?